### PR TITLE
ELPP-3561 Append query parameters to redirect uri

### PR DESCRIPTION
**Pull request by nlisgo**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
I need more insight on the goal of this, because at the moment Journal is behind a CDN and hence may not see query parameters, if they are stripped away at the edge?

**Comment by thewilkybarkid** [comment]
They're not, we use some parameters (eg `page`). `https://elifesciences.org/` `https://elifesciences.org/?foo` give a different `Age`.

**Comment by thewilkybarkid** [comment]
(Though it would be a good idea to whitelist them.)

**Comment by thewilkybarkid** [comment]
(Also a good idea to do redirects at the edge, but requires Fastly.)

**Comment by giorgiosironi** [comment]
> They're not, we use some parameters (eg page). https://elifesciences.org/ https://elifesciences.org/?foo give a different Age.

Viability is ok then, I'm testing the implementation now.

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 1]
```diff
@@ -74,7 +74,7 @@ server {
     rewrite '^/Labs$' '/labs' permanent;
```
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-available-journal.conf]: these redirects seem to already preserve the query string:
```
curl -v "https://elifesciences.org/About?p1=v1&p2=v2"
...
< Location: http://elifesciences.org/about?p1=v1&p2=v2
```
```
curl -v "https://elifesciences.org/content/5/e10627?p1=v1&p2=v2"
...
< Location: http://elifesciences.org/articles/10627?p1=v1&p2=v2
```
Are there other scenarios that are broken?

**Comment by nlisgo** [comment]
This is related to ELPP-3513.

It would be acceptable for links from `journal-cms` to `journal` not to contain the slug as long as we can preserve any campaign tracking at the end of URLs.

**Comment by giorgiosironi** [comment]
> as long as we can preserve any campaign tracking at the end of URLs

From my tests (see inline comment) the nginx redirects seem to preserve the campaign tracking and other query string parameters

**Comment by giorgiosironi** [comment]
Discussed on call, this can likely be closed as already implemented.

**Pull request closed without merging**